### DISCUSSION TITLE
Ability to warn instead of throw when reaching max requests per session

### DISCUSSION THREAD

**Discussion by loraderon** [ravendb/ravendb#19815] [Ideas]
It would be great if we were able to configure a softer version of `MaxNumberOfRequestsPerSession` that only logs warnings (to default logger and/or server alerts).
This way we can have a higher (not really wanted) value for `MaxNumberOfRequestsPerSession` and keep an eye out for sessions that exceed the value for `MaxNumberOfRequestsPerSessionWarningLimit`.

It must probably be nullable (and null by default) and if either `MaxNumberOfRequestsPerSessionWarningLimit` or `MaxNumberOfRequestsPerSession` is set it must be validated that the warning limit is lower than `MaxNumberOfRequestsPerSession` or `null`.

**Comment by karmeli87** [comment]
Hi,

We expose the `NumberOfRequests` that are done per session, so you can do something like this:

```
session.Advanced.OnAfterSaveChanges += (sender, args) =>
{
	if (args.Session.NumberOfRequests > 100)
	{
		// log
	}
};
```

**Reply by loraderon** [reply]
That will work perfectly for my use case, thank you!
